docs/features.md

```diff
@@ -0,0 +1,408 @@
+
+### Steps
+
+```
+# MyFirstTestCase
+
+require browserPath
+
+* Open application and navigate to the question of life
+  
+  Component: WebBrowser
+  - Open browser @browserPath
+  - Navigate to "http://org.question-of-life/index.html"
+
+* Enter the question
+
+  Component: QOLPage
+  - Enter "What's the meaning of life?" into <QuestionField>
+  - Click <AskButton>
+  
+* Validate the correct answer to be _42_
+
+  Component: QOLPage
+  - answer = Read <AnswerField>
+  - assert answer = "42"
+```
+
+Now each specification step is *implemented* using appropriate test steps. Each test step is valid only within a context. The context can be
+either a Component / Mask or a Macro. Component and Mask can be used synonymously whereas Macros are somewhat different and will be
+discussed later on.
+
+A Component is an element of the application and is defined via the aml. I will make use of Components without going into to much detail for
+now. Components will be fully discussed in the AML-Section.
+
+#### Test Step
+
+`- Open browser @browserPath` is an example of a Test Step. It is a concise description of what should happen that will map directly onto an
+implementation. Each test step maps exactly onto one fixture call. In this case the fixture is called with one (String) parameter
+`browserPath`. This parameter is actually an environment variable (which I will discuss in detail later on). Variables can generally be used
+to hold values of type `long`/`Long`, `boolean`/`Boolean`, `String` and `Map<String,Object>`. They can be passed to fixture calls, macro
+calls and can be used in assertions.
+
+#### Assignment Step
+
+`- answer = Read <AnswerField>` is an Assignment Step, that makes a call to a fixture using `AnswerField` as *element* parameter (element
+parameters are discussed in the application mapping), returning a value that is assigned to `answer`. The assigned variable is accessible
+from this statement on until the end of this file. Each variable can only be assigned once. What type the assigned variable actually has
+depends on the return type of the fixture that is effectively called. The Test-Editor makes sure that the types are checked before running
+the test such that a user should not have to worry about types until type-violations cannot be solved.
+
+#### Assertion Step
+
+`- assert answer = "42"` will fail if the variable `answer` does not hold 42. Assertions are used to specify expectations in a test
+case. The expression that follows the assertion cannot make use of recursively nested expression. The following expression are allowed
+* simple boolean check (e.g. `- assert boolVar` or `- assert !boolVar` for negation)
+* simple comparitive check using operators `=`, `!=`, `>=`, `>`, `<`, `<=` (e.g. `- assert answer = "42"`)
+* string matching operations (e.g. `- someString matches "^notmuch.*of.+an$"` or `- someString does not match "[A-Z]+"`)
+* values can be either variables or string constants
```

Comment:
This will have to be fixed in the generator (generating the appropriate org.junit.Assert method calls).